plugins/woocommerce/src/Internal/DataStores/CustomMetaDataStore.php

```diff
@@ -232,4 +232,35 @@ public function get_metadata_by_key( &$object, string $meta_key ) {
 		return $meta;
 	}
 
+	/**
+	 * Returns distinct meta keys in use.
+	 *
+	 * @since 8.8.0
+	 *
+	 * @param int    $limit           Maximum number of meta keys to return. Defaults to 100.
+	 * @param string $order           Order to use for the results. Either 'ASC' or 'DESC'. Defaults to 'ASC'.
+	 * @param bool   $include_private Whether to include private meta keys in the results. Defaults to FALSE.
+	 * @return string[]
+	 */
+	public function get_meta_keys( $limit = 100, $order = 'ASC', $include_private = false ) {
+		global $wpdb;
+
+		$db_info = $this->get_db_info();
+
+		$query = "SELECT DISTINCT meta_key FROM {$db_info['table']} ";
+
+		if ( ! $include_private ) {
+			$query .= $wpdb->prepare( 'WHERE meta_key NOT LIKE %s ', $wpdb->esc_like( '_' ) . '%' );
+		}
+
+		$order  = in_array( $order, array( 'ASC', 'DESC' ), true ) ? $order : 'ASC';
```

Comment:
Should probably normalize `$order` with `strtoupper` before doing this check so that `asc` and `desc` will work as well.

Reply:
Good idea. Done in 7f2198da910cd8a5d125214400d9b628049cd404.